compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2213,16 +2213,38 @@ class Typer extends Namer
             traverse(xtree :: rest, facts)
           case none =>
             import untpd.modsDeco
-            val ctx1 = if (ctx.explicitNulls) {
-              mdef match {
-                case mdef: untpd.ValDef if ctx.owner.is(Method) && !mdef.mods.isOneOf(Lazy | Implicit) =>
-                  ctx.fresh.addFlowFacts(facts)
-                case _ => ctx
+            // Force completer to use calling context (as opposed to the creation context)
+            // to complete itself. This approach is used so that flow typing can handle `ValDef`s
```

Comment:
update comment

"can handle `ValDefs` that appear within a block"

=>

"can handle definitions appearing within a block."